riscv/processor.h

```diff
@@ -283,7 +283,7 @@ class processor_t : public abstract_device_t
     return impl_table[impl];
   }
   reg_t pc_alignment_mask() {
-    return ~(reg_t)(extension_enabled('C') ? 0 : 2);
+    return ~(reg_t)(extension_enabled('C') || extension_enabled(EXT_ZCA)  ? 0 : 2);
```

Comment:
After making my suggested change to implicitly enable Zca when C is present, this should be simplified to check only Zca.